docs/UpgradeGuide.md

```diff
@@ -171,11 +219,82 @@ end
+| `Datadog.tracer.active_span.context`                                                         | Removed                                     | Consider `Datadog::Tracing.active_trace` instead.                   |
+| `Datadog.tracer.active_span`                                                                 | `Datadog::Tracing.active_span`              | See [trace state](#1.0-trace-api-trace-state) for details.          |
+| `Datadog.tracer.call_context`                                                                | Removed                                     | See [trace state](#1.0-trace-api-trace-state) for details.          |
+| `Datadog.tracer.configure(options)`                                                          | `Datadog.configure { \|c\| ... }`           | Use configuration API instead.                                      |
+| `Datadog.tracer.provider.context = context`                                                  | `Datadog::Tracing.continue_from!(digest)`   | See [distributed tracing](#1.0-trace-api-distributed) for details.  |
+| `Datadog.tracer.set_tags(tags)`                                                              | `Datadog.configure { \|c\| c.tags = tags }` |                                                                     |
+| `Datadog.tracer.shutdown!`                                                                   | `Datadog::Tracing.shutdown!`                |                                                                     |
+| `Datadog.tracer.start_span`                                                                  | `Datadog::Tracing.trace`                    | See [manual tracing](#1.0-trace-api-manual-tracing) for details.    |
+| `Datadog.tracer.trace`                                                                       | `Datadog::Tracing.trace`                    | See [manual tracing](#1.0-trace-api-manual-tracing) for details.    |
+
+
+Also check out the functions defined within `Datadog::Tracing` in our [public API](https://www.rubydoc.info/gems/ddtrace/) for more details on their usage.
+
+<h3 id="1.0-trace-api-removed-tracer">Removed `Datadog.tracer`</h3>
+
+Many of the functions accessed directly through `Datadog.tracer` have been moved to `Datadog::Tracing` instead.
+
+<h3 id="1.0-trace-api-removed-context">Removed access to `Datadog::Context`</h3>
 
-Manual tracing is now done through the public API.
+Direct usage of `Datadog::Context` has been removed. Previously, it was used to modify or access active trace state. Most use cases have been replaced by our [public trace API](https://www.rubydoc.info/gems/ddtrace/).
 
-Whereas in 0.x, the block would yield a `Datadog::Span` as `span`, in 1.0, the block yields a `Datadog::SpanOperation` as `span` and `Datadog::TraceOperation` as `trace`.
+<h3 id="1.0-trace-api-manual-tracing">Manual tracing & trace model</h3>
+
+Manual tracing is now done through the [public API](https://www.rubydoc.info/gems/ddtrace/).
```

Comment:
The rubydoc.info site hasn't yet been updated with 1.0.0.beta1 (probably only indexes non-beta versions?) but I think it would be nice to update this document after release with deeper links to specific classes/parts of the API. (So not for this PR, just a suggestion for a later improvement)